### ISSUE TITLE
ProxyFactory#getProxy seems to re-use EVM instances from the cache without properly settings optionalParameters

### ISSUE THREAD

**Issue by EugenMayer** [Blazebit/blaze-persistence#1805]
Assuming that one uses `entityViewConfiguration.addOptionalParameters` to insert default parameters for an EVM to carry (specifically beans on needs to use during Pre/Post remove listeners), it seems that during the invocation attempt for thus listeners, and EVM instance is created via `context.getEntityViewManager().getSerializableDelegate(view.$$_getEntityViewClass())`
 which ultimately will use the `ProxyFactory` to create or **reuse* an EVM.

When an EVM is "reused" it is not ensured, that this EVM has the optionalParameters applied and thus we can end up having an EVM without those parameters included.

**Comment by Mobe91** [comment]
@EugenMayer would you be able to try if the changes in https://github.com/Blazebit/blaze-persistence/pull/1806 solve this?

**Comment by EugenMayer** [comment]
@Mobe91 thank you a lot for working on this item! Will try to queue the test today

**Comment by EugenMayer** [comment]
@Mobe91 indeed this solves my issue entirely. Did it solve your memory leak issue too?

**Comment by Mobe91** [comment]
> Did it solve your memory leak issue too?

Unfortunately not, there must be other things going on in my case that are probably not related to blaze-persistence 🥲

Glad to hear that it works for you! @beikov will make sure it goes into the upcoming release.

**Comment by EugenMayer** [comment]
@Mobe91 AFAIK you are using spring boot in your tests right? 

Some aspects we use:
1. I'am sure you are aware about the context cache and the ability to limit the amount of caches spring boot is preserving during the tests? For us, this has limitted the used memory and connections. `spring.test.context.cache.maxSize=5` or similar helps our cause.

2. In addition, you might want to implement an TestExecutionListener that clears the EM L1 cache after every test or test suite.

3. Also, if you are using gradle you might tune your test task

```
    maxParallelForks = 1
    systemProperties['junit.jupiter.execution.parallel.enabled'] = false
```

We use this for all JPA based tests.

**Comment by Mobe91** [comment]
Thank you for the hints. If you are curious, I've opened a [spring-boot issue](https://github.com/spring-projects/spring-boot/issues/37129) a while ago for this and included quite some details about what I am observing in my case. Reducing the application context cache size was the first thing I did but strangely the application contexts are still not garbage collected even though they are no longer held by the cache.

My tests run in Maven but no parallelism is involved. I might give the EM clearing a try though! 👍
